controller/src/constants.rs

```diff
@@ -15,3 +17,20 @@ pub(crate) fn requeue_slow() -> Action {
 pub(crate) fn no_requeue() -> Action {
     Action::await_change()
 }
+
+/// Parse a duration string into a Duration object.
+pub(crate) fn parse_duration(input: &str) -> Result<Duration, ParseIntError> {
+    let mut secs: u64 = 0;
+    let mut duration_string = input;
+    let units = vec![('d', 86400), ('h', 3600), ('m', 60)];
```

Comment:
Have you verified that d,h, and m are the only possible units for this data format?

Reply:
d, h, m, and s are the only characters that can be in the string as per the validation regex. Since s (if present) is guaranteed to be the last character and the associated value would be 1, I am not including that in the vector.